consensus/replay_test.go

```diff
@@ -278,42 +281,194 @@ var (
 // 2 - save block and committed but state is behind
 var modes = []uint{0, 1, 2}
 
+// This actually is not a test, it's for storing validator change tx data for testHandshakeReplay
+func TestSimulateValidatorsChange(t *testing.T) {
+	nPeers := 7
+	nVals := 4
+	css, genDoc, config, cleanup := randConsensusNetWithPeers(nVals, nPeers, "replay_test", newMockTickerFunc(true), newPersistentKVStoreWithPath)
```

Comment:
Oy. It's so bad that so much complexity is required just to simulate the creation of a valid blockchain. We'll need to address this. Thanks for working through it though.

Reply:
Do you mean I can just hardcode some block data with txs changing validators, what if in the future we change some format of block or tx? then we are going to hardcode those blocks again?

Reply:
Em, that could be some better way. But hardcode is clearly not feasible.

Reply:
No, not hard code, but it should be much easier to build a valid blockchain with the functions. No need to go through the entire consensus process like this!

Reply:
I use a new test case to simulate the chain, just start one consensusState, the others are mocked, like the style in state.go.  The time executing it is about 0.25s now.